hasura/metadata/databases/default/tables/public_vaults.yaml

```diff
@@ -23,9 +23,12 @@ insert_permissions:
       protocol:
         circles:
           users:
-            profile:
-              id:
-                _eq: X-Hasura-User-Id
+            _and:
+            - deleted_at:
+                _is_null: true
+            - profile:
+                id:
+                  _eq: X-Hasura-User-Id
```

Comment:
not too sure but would it expose any security issue if any random member of the circle use the api directly to create a vault record ?

Reply:
good point! we should really disable this and write a custom mutation.

Reply:
@teeolendo this is an interesting one... i think we can still pass the vault address from the frontend when creating a vault, but the server will need to validate it. otherwise a circle member could create a "vault" with the address of a malicious contract

Reply:
if there is a vault registry i guess you can pass some sort of id or index instead of address to prevent address spoofing

Reply:
This issue predates the changes in this PR, so I'm not going to consider it a merge blocker, but I'll make a task for it